src/base/bittorrent/trackerentry.cpp

```diff
@@ -52,59 +52,56 @@ QString TrackerEntry::url() const
     return QString::fromStdString(nativeEntry().url);
 }
 
-bool TrackerEntry::isWorking() const
-{
-    // lt::announce_entry::is_working() returns
-    // true when the tracker hasn't been tried yet.
-#if (LIBTORRENT_VERSION_NUM < 10200)
-    return nativeEntry().verified && nativeEntry().is_working();
-#else
-    if (!nativeEntry().verified)
-        return false;
-    const auto &endpoints = nativeEntry().endpoints;
-    return std::any_of(endpoints.begin(), endpoints.end()
-                       , [](const lt::announce_endpoint &endpoint)
-    {
-        return endpoint.is_working();
-    });
-#endif
-}
-
 int TrackerEntry::tier() const
 {
     return nativeEntry().tier;
 }
 
 TrackerEntry::Status TrackerEntry::status() const
 {
-    if (isWorking())
-        return Working;
-
 #if (LIBTORRENT_VERSION_NUM < 10200)
-    if ((nativeEntry().fails == 0) && nativeEntry().updating)
+    if (nativeEntry().fails > 0)
+        return NotWorking;
+
+    if (nativeEntry().updating)
         return Updating;
-    if (nativeEntry().fails == 0)
-        return NotContacted;
 #else
     const auto &endpoints = nativeEntry().endpoints;
-    const bool allFailed = std::all_of(endpoints.begin(), endpoints.end()
-                                       , [](const lt::announce_endpoint &endpoint)
+
+    const bool allFailed = !endpoints.empty() && std::all_of(endpoints.begin(), endpoints.end()
+        , [](const lt::announce_endpoint &endpoint)
     {
         return (endpoint.fails > 0);
     });
-    const bool updating = std::any_of(endpoints.begin(), endpoints.end()
-                                      , [](const lt::announce_endpoint &endpoint)
+    if (allFailed)
+        return NotWorking;
+
+    const bool isUpdating = std::any_of(endpoints.begin(), endpoints.end()
+        , [](const lt::announce_endpoint &endpoint)
     {
         return endpoint.updating;
     });
-
-    if (!allFailed && updating)
+    if (isUpdating)
         return Updating;
+#endif
 
-    if (!allFailed)
+    if (!nativeEntry().verified)
         return NotContacted;
+
+#if (LIBTORRENT_VERSION_NUM < 10200)
+    if (nativeEntry().is_working())
```

Comment:
You don't really need to check this condition. `is_working()` just returns `fails == 0;` so it's constantly `true` here.
Assertion below is also redundant.

Reply:
OK, updated.